mono/eglib/glib.h

```diff
@@ -1145,6 +1145,20 @@ glong     g_utf8_pointer_to_offset (const gchar *str, const gchar *pos);
 #define G_HAVE_API_SUPPORT(x) (x)
 #define G_UNSUPPORTED_API "%s:%d: '%s' not supported.", __FILE__, __LINE__
 #define g_unsupported_api(name) G_STMT_START { g_warning (G_UNSUPPORTED_API, name); } G_STMT_END
+
+#ifdef ENABLE_ICALL_EXPORT
+#pragma GCC diagnostic ignored "-Wmissing-prototypes"
+#define ICALL_DECL_EXPORT MONO_API
```

Comment:
This will work with a strike of luck though, since you are effectively using a `#define` that you can only hope will be defined later, thus introducing a intrinsic dependency between `glib.h` and `mono-compiler.h` (where `MONO_API` is defined IIRC). Why moving these `ICALL_DECL_EXPORT` and `ICALL_EXPORT` to `glib.h` where it has nothing to do?

Defining it in `icall.h` which can then be included everywhere you need it will fix the issue at hand here.

Reply:
It isn't quite luck, in that we check, so if it isn't defined later we fail (at least if configured correctly).

It was previously in class-internals.h. I wanted it somewhere where it'd be widely available, but not public, and not add much new exposure, i.e. not add more include class-internals.h.

glib.h is probably the most widely available non-public header. There is no `icall.h`. Do you mean `icalls.h` that I had earlier? Or introduce a new file icall.h that include publib please these lines?

Reply:
I’ll add icall.h and a bunch of includes. Utils since it is needes there, unless we moving them all to metadata.